### PR TITLE
Implement `return_indices` option to `F.max_pooling_nd`

### PR DESCRIPTION
**Pull request by kmaehashi**
Refs https://github.com/chainer/chainer/pull/4421#issuecomment-395377917

### CODE REVIEW

**Comment by hvy** [comment]
Merge https://github.com/chainer/chainer/pull/4890 first, since they are very similar.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Updated for comments in #4890.

**Comment by hvy** [comment]
jenkins, test this please.

File: chainer/functions/pooling/max_pooling_nd.py [hunk 1]
```diff
@@ -257,10 +258,26 @@ def max_pooling_nd(x, ksize, stride=None, pad=0, cover_all=True):
...
+            expected for use by :func:`chainer.functions.upsampling_nd`.
+            Note that cuDNN will not be used for this function if
+            ``return_indices`` is set to ``True``, as cuDNN does not return
+            indices information.
 
     Returns:
-        ~chainer.Variable: Output variable.
+        ~chainer.Variable or tuple:
+            When ``return_indices`` is ``False`` (default), returns the output
+            variable. Otherwise returnes the tuple of the output variable and
```
Comment (hvy) [inline chainer/functions/pooling/max_pooling_nd.py]: Could you address https://github.com/chainer/chainer/pull/4890#discussion_r195967131 here too?

Comment (kmaehashi) [reply chainer/functions/pooling/max_pooling_nd.py]: Fixed.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**